### PR TITLE
added React/week2 homework

### PR DESCRIPTION
**Pull request by DeepthiDukka**

### CODE REVIEW

File: react/week2/todo-list-app/src/components/Timer.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: 0,
+    };
+    this.timer = this.timer.bind(this);
+  }
+
+  componentDidMount() {
+    this.counter = setInterval(this.timer, 1000);
```
Comment (benna100) [inline react/week2/todo-list-app/src/components/Timer.js]:
```suggestion
    this.setIntervalId= setInterval(this.timer, 1000);
```

File: react/week2/todo-list-app/src/components/Timer.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  componentWillUnmount() {
+    clearInterval(this.counter);
```
Comment (benna100) [inline react/week2/todo-list-app/src/components/Timer.js]: Well done!

File: react/week2/todo-list-app/src/components/TodoListItem.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      {
+        id: 3,
+        description: "Eat breakfast",
+        checked: false,
+      },
+    ],
+    todoAddNewItem: " ",
+  };
+
+  addTodo = (todoAddNewItem) => {
```
Comment (benna100) [inline react/week2/todo-list-app/src/components/TodoListItem.js:25]: This variable name can be improved todoAddNewItem

File: react/week2/todo-list-app/src/components/TodoListItem.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+          id: 4,
+          name: todoAddNewItem,
+          description: "random text",
+          completed: false,
+        },
+      ],
+    });
+  };
+
+  deleteTodo = (index) => {
```
Comment (benna100) [inline react/week2/todo-list-app/src/components/TodoListItem.js]: If i made a function that deleted todoitems i would make sure its easy to use the function. Index could mean lost of things, but i guess you mean id? Or do you mean the index it is in an array or in the rendered list?

```suggestion
  deleteTodo = (itemId) => {
```

File: react/week2/todo-list-app/src/components/TodoListItem.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+          handleCheckbox={this.handleCheckbox}
+        />
+      </div>
+    );
+  }
+}
+
+class Todo extends Component {
+  render() {
+    if (this.props.todoList.length === 0) return <p>No todo items</p>;
```
Comment (benna100) [inline react/week2/todo-list-app/src/components/TodoListItem.js:77]: great to check for this first

File: react/week2/todo-list-app/src/components/TodoListItem.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    let result = this.props.todoList.map((item, index) => {
```
Comment (benna100) [inline react/week2/todo-list-app/src/components/TodoListItem.js]: result is a vague variable name, can you improve it?

**Review by benna100 — COMMENTED** [review]
Good job Deepthi :) Try and focus more on giving good variable names :)

File: react/week2/todo-list-app/src/components/TodoListItem.js (hunk 6 above)
Comment (DeepthiDukka) [reply react/week2/todo-list-app/src/components/TodoListItem.js]: implemented the feedback as per the above suggestions.

**Pull request merged**